pkg/platform/runtime/store/store.go

```diff
@@ -80,33 +79,33 @@ func (s *Store) MarkerIsValid(commitID strfmt.UUID) bool {
 		return false
 	}
 
-	f, err := os.OpenFile(marker, os.O_RDONLY, 0)
+	contents, err := fileutils.ReadFile(marker)
 	if err != nil {
-		logging.Error("Could not open marker file: %v", err)
+		logging.Error("Could not read marker file %s: %v", marker, err)
+	}
+	lines := strings.Split(string(contents), "\n")
+	if len(lines) < 1 {
+		logging.Debug("Expected commit ID in marker file, but was empty")
 		return false
 	}
-	defer f.Close()
-
-	sc := bufio.NewScanner(f)
-	ok := sc.Scan()
-	if !ok {
-		logging.Error("Expected commit ID in marker file: %v", sc.Err())
+	parsedCommitID := strings.TrimSpace(lines[0])
+	if len(lines) < 2 {
+		logging.Debug("Expected State Tool version in marker file")
 		return false
 	}
-	parsedID := strings.TrimSpace(sc.Text())
-	logging.Debug("Matching commitID for %s, %s==%s", marker, parsedID, commitID.String())
-	if parsedID != commitID.String() {
+	parsedStateToolVersion := strings.TrimSpace(lines[1])
+
+	if parsedCommitID != commitID.String() {
+		logging.Debug("Could not match commitID in %s, expected: %s, got: %s", marker, commitID.String(), parsedCommitID)
 		return false
 	}
 
-	ok = sc.Scan()
-	if !ok {
-		logging.Debug("Expected State Tool version in marker file: %v", sc.Err())
+	if parsedStateToolVersion != fmt.Sprintf("%s@%s", constants.BranchName, constants.Version) {
```

Comment:
Didn't notice this the first review; we should not include the branch name. Just the State Tool version. If we do our job right the version should be the same regardless of which branch it's on.

Reply:
the sha should always differ, that is true.